pkg/cmd/subcommand/demo.go

```diff
@@ -0,0 +1,270 @@
+package subcommand
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"os"
+	"runtime"
+	"strings"
+	"time"
+
+	"github.com/OpenFunction/cli/pkg/client"
+	"github.com/OpenFunction/cli/pkg/components/common"
+
+	"github.com/OpenFunction/cli/pkg/cmd/util"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"golang.org/x/sync/errgroup"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	k8s "k8s.io/client-go/kubernetes"
+)
+
+// Demo is the commandline for 'init' sub command
+type Demo struct {
+	genericclioptions.IOStreams
+
+	Verbose             bool
+	RegionCN            bool
+	OpenFunctionVersion string
+	DryRun              bool
+	Delete              bool
+	Timeout             time.Duration
+}
+
+func init() {
+	w = os.Stdout
+	availableVersions = []string{"v0.3.1", "v0.4.0", "latest"}
+}
+
+// NewDemo returns an initialized Init instance
+func NewDemo(ioStreams genericclioptions.IOStreams) *Demo {
+	return &Demo{
+		IOStreams: ioStreams,
+	}
+}
+
+func NewCmdDemo(restClient util.Getter, ioStreams genericclioptions.IOStreams) *cobra.Command {
+	var cl *k8s.Clientset
+
+	i := NewDemo(ioStreams)
+
+	cmd := &cobra.Command{
+		Use:                   "demo <args>...",
+		DisableFlagsInUseLine: true,
+		Short:                 "Create OpenFunction demo.",
+		Long: `
+This command will help you to Run the OpenFunction demo.
+
+You can use fn demo  to run the OpenFunction demo.
+
+The demo running is OpenFunction v0.4.0 are: Dapr, KEDA, Knative Serving, Shipwright, Tekton Pipelines.
+
+The permitted parameters are: --delete, --region-cn, --version, --verbose, --dry-run
+
+The dependencies to be installed for OpenFunction v0.4.0 are: Dapr, KEDA, Knative Serving, Shipwright, Tekton Pipelines, Cert Manager, Ingress Nginx.
+
+The permitted parameters are: --delete, --region-cn, --version, --verbose, --dry-run
+`,
+		Example: "fn demo",
+		Run: func(cmd *cobra.Command, args []string) {
+			if i.Delete {
+				util.CheckErr(i.DeleteKind(cmd, args))
+			} else {
+				util.CheckErr(i.RunKind(cl, cmd))
+				installCMD := NewCmdInstall(restClient, ioStreams)
+
+				installCMD.SetArgs([]string{
+					fmt.Sprintf("install"),
+					fmt.Sprintf("--all"),
+					fmt.Sprintf("--region-cn=%v", i.RegionCN),
+					fmt.Sprintf("--version=latest")},
+				)
+				installCMD.Execute()
+				util.CheckErr(i.RunDemo(cl, cmd))
+			}
+		},
+	}
+
+	cmd.Flags().BoolVar(&i.Verbose, "verbose", false, "Show verbose information.")
+	cmd.Flags().BoolVar(&i.RegionCN, "region-cn", false, "For users in China to uninstall dependent components.")
+	cmd.Flags().BoolVar(&i.DryRun, "dry-run", false, "Used to prompt for the components and their versions to be uninstalled by the current command.")
```

Comment:
this is for demo, not uninstall.
And do we need a dry-run for demo? @wenchajun @tpiperatgod